wsidicom/instance/dataset.py

```diff
@@ -378,13 +398,13 @@ def image_size(self) -> Size:
             raise WsiDicomError("Image size is zero")
         if self.tile_type == TileType.FULL:
             expected_tiled_size = image_size.ceil_div(self.tile_size)
-            if expected_tiled_size.area != self.frame_count:
+            if expected_tiled_size.area != self.tile_frame_count:
```

Comment:
I suggest to change the check to include the focal planes and optical paths here like:
```
            number_of_focal_planes = int(getattr(self, "TotalPixelMatrixFocalPlanes", 1))
            number_of_optical_paths = int(getattr(self, "NumberOfOpticalPaths", 1))
            if (
                expected_tiled_size.area
                * number_of_focal_planes
                * number_of_optical_paths
                != self.tile_frame_count
            ):
```
This way we do not need to introduce the new properties that are problematic if used on sparse datasets.

Reply:
Note that we will likely need to add this logic as well for [this line](https://github.com/imi-bigpicture/wsidicom/pull/116/files#diff-c44675a399dd56697e2e27a039efc173fc21eba3d17cb0ebd20a73d6a9e59b4eR274). That is why I originally implemented it as a property.